### DISCUSSION TITLE
Plugin sample

### DISCUSSION THREAD

**Discussion by jonasraoni** [pkp/pkp-lib#9116] [Proposals]
In order to standardize the plugin development practices, and also to help users starting from zero, it makes sense to have a sample plugin with everything that might be needed.

**Infrastructure**
- Versioning/branch pattern
- Build: GitHub Actions or Travis?
- Unit tests
- Cypress tests
- Code formatter: pre-commit or auto-format after pushing?
- Tools (PHPUnit, PHPStan, ...)
- Deployment/release script (considering the existence of composer packages)

**Possible code samples**
- Using jobs
- Setup plugin settings
- Notifications
- Email
- Listening to hooks/events
- Extend the API
- Overwrite something from the Laravel container
- Database interaction
- Creation of custom schema/tables
- Upgrade/migration script
- Extending the reader/manager interface
- Reusing/extending Vue components at the manager interface

**Comment by asmecher** [comment]
@jonasraoni, you might want to take a look at https://github.com/jardakotesovec/backend-ui-example-plugin, which @jardakotesovec is building into a VueJS example.

**Reply by jonasraoni** [reply]
I'll take a look soon, I'm getting back on track 😄
As part of this discussion, do you think I should add "3rd-parties should have a standard way to notify users about new versions of their plugins"? 🤔

**Reply by asmecher** [reply]
I haven't thought about that much -- I'm interested in possible approaches!
